docs/anti-patterns.mdx

```diff
@@ -4,35 +4,111 @@ title: Cadence Anti-Patterns
 
-### Named Value Fields Are Preferable
+## Events From Resources Might Not Be Unique
 
-Where contracts, resources, and scripts all have to repeatedly refer to the same constant values, entering these manually is a potential source of error. These are known as [magic values](<https://en.wikipedia.org/wiki/Magic_number_(programming)>).
+### Problem
 
-Rather than typing the values manually in functions, transactions and scripts, store the values once in fields of the contract responsible and then access them via those fields. This is the [Named Value Field pattern](design-patterns/#named-value-field)
+A public function on a contract that creates a resource can be called by any account.
+If that resource has functions that emit events, any account can create an instance of that resource and emit those events.
+If those events are meant to indicate actions taken using a single instance of that resource
+(for example an admin object, switchboard, or registry),
+or instances created through a particular workflow, then this will make event log search
+and management harder because events from other instances may be mixed in with the ones that you wish to examine.
+
+### Solution
+
+To fix this, if there should be only a single instance of the resource,
+it should be created and `link()`ed to a public path in an admin account's storage
+during the contracts's initializer.
 
-### Public Functions and Fields Should Be Avoided
+# Access Control
+
+## Public Functions and Fields Should Be Avoided
+
+### Problem
 
 Be sure to keep track of access modifiers when structuring your code, and make public only what should be public.
+Accidentally exposed fields can be a security hole.
 
-### Array or Dictionary Fields Should Be Private
+### Solution
 
-#### Problem
+When writing your smart contract, look at every field and function and make sure
+that they are all `access(self)`, `access(contract)`, or `access(account)`, unless otherwise needed.
+
+## Capability-Typed Public Fields Are A Security Hole
 
 This is a specific case of "Public Functions And Fields Should Be Avoided", above.
 
-Public array or dictionary fields are not directly over-writable, but their members can be accessed and overwritten if the field is public. This could potentially result in security vulnerabilities for the contract if these fields are mistakenly made public.
+### Problem
+
+The values of public fields can be copied. Capabilities are value types,
+so if they are used as a public field, anyone can copy it from the field
+and call the functions that it exposes.
+This almost certainly is not what you want if a capability
+has been stored as a field on a contract or resource in this way.
+
+### Solution
+
+For public access to a capability, place it in an accounts public area so this expectation is explicit.
+
+## Array or Dictionary Fields Should Be Private
+
+<Callout type="info">
+
+This anti-pattern will no longer be an issue after the April 2022 Flow Spork and network upgrade.
```

Comment:
This breaking change will not get deployed in that spork, but an upcoming one (likely the following one). I would just point to the FLIP and not mention a date